### PR TITLE
Deflate axis

### PR DESCRIPTION
**Pull request by couet**
fix https://github.com/root-project/root/issues/8598
It is reasonable to call automatically LabelsDeflate at drawing time when plotting histogram with labels.

### CODE REVIEW

File: hist/histpainter/src/THistPainter.cxx [hunk 1]
```diff
@@ -4490,6 +4490,10 @@ void THistPainter::Paint(Option_t *option)
       return;
    }
 
+   // Deflate the labels in case of alphanumeric labels
```
Comment (lmoneta) [inline hist/histpainter/src/THistPainter.cxx:4493]: The correct condition is:
```
if (fXaxis.CanExtend() && fXaxis.IsAlphanumeric()) fH->LabelsDeflate("X");
if (fYaxis.CanExtend() && fYaxis.IsAlphanumeric()) fH->LabelsDeflate("Y");
```

and should we add also the case for the Z axis (for TH3 objects) ?

**Review by lmoneta — CHANGES_REQUESTED** [review]
Thank you Olivier for the PR. 
Checking on the fact that an axis can be extended and is alphanumeric is the correct thing, because you can have histograms with labels and the users explicitly does not want the axis to be inflated/deflated.

**Comment by couet** [comment]
@lmoneta 
Hello Lorenzo, I implemented your suggestions. I will merge as soon as you approve it.

**Review by lmoneta — APPROVED** [review]
LGTM now ! 
Thank you Olivier for the update !

**Pull request merged**